napalm_ios/ios.py

```diff
@@ -582,19 +582,24 @@ def get_lldp_neighbors(self):
             try:
                 device_id, local_int_brief, hold_time, capability, remote_port = lldp_entry.split()
             except ValueError:
-                # Check for long names (device_id is 20 chars long)
                 if len(lldp_entry.split()) == 4:
-                    tmp_field, hold_time, capability, remote_port = lldp_entry.split()
-                    device_id = tmp_field[:20]
-                    local_int_brief = tmp_field[20:]
-                    # device_id might be abbreviated, try to get full name
-                    lldp_tmp = self._lldp_detail_parser(local_int_brief)
-                    device_id_new = lldp_tmp[3][0]
-                    # Verify abbreviated and full name are consistent
-                    if device_id_new[:20] == device_id:
-                        device_id = device_id_new
+                    # Four fields might be long_name or missing capability
+                    capability_missing = True if lldp_entry[46] == ' ' else False
```

Comment:
Short answer...   F****ing Cisco

Longer answer...Cisco merged two fields together (no spaces between them) so it is really hard to parse. Looking at the output, it looks like the field widths are fixed number of spaces apart and the 'capability' field starts at 47th character (i.e. index 46). If there is nothing, there then the capability is missing.